internal/eval/store_eval.go

```diff
@@ -910,3 +910,62 @@ func evalPFMERGE(args []string, store *dstore.Store) *EvalResponse {
 		Error:  nil,
 	}
 }
+
+// ZPOPMIN Removes and returns the member with the lowest score from the sorted set at the specified key.
+// If multiple members have the same score, the one that comes first alphabetically is returned.
+// You can also specify a count to remove and return multiple members at once.
+// If the set is empty, it returns an empty result.
+func evalZPOPMIN(args []string, store *dstore.Store) *EvalResponse {
+	// Incorrect number of arguments should return error
+	if len(args) < 1 {
+		return &EvalResponse{
+			Result: clientio.NIL,
+			Error:  diceerrors.ErrWrongArgumentCount("ZPOPMIN"),
+		}
+	}
+
+	key := args[0]        // Key argument
+	obj := store.Get(key) // Getting sortedSet object from store
+
+	// If the sortedSet is nil, return an empty list
+	if obj == nil {
+		return &EvalResponse{
+			Result: []string{},
+			Error:  nil,
+		}
+	}
+
+	sortedSet, err := sortedset.FromObject(obj)
+	if err != nil {
+		return &EvalResponse{
+			Result: clientio.NIL,
+			Error:  diceerrors.ErrWrongTypeOperation,
+		}
+	}
+
+	var results []string
+
+	// If the count argument is not passed, return the first member with lowest score
+	if len(args) < 2 {
+		results = sortedSet.GetMin(1)
+	} else {
+		count, err := strconv.Atoi(args[1])
+
+		// If the count argument is invalid, return integer error
+		if err != nil {
+			return &EvalResponse{
+				Result: clientio.NIL,
+				Error:  diceerrors.ErrIntegerOutOfRange,
+			}
+		}
+
+		// If the count argument is present, return all the members with lowest score sorted in ascending order.
+		// If there are multiple lowest scores with same score value, it sorts the members in lexographical order of member name
+		results = sortedSet.GetMin(count)
+	}
```

Comment:
This may be a more readable way to structure this.
```suggestion
count := 1

// Check if the count argument is provided
if len(args) >= 2 {
	countArg, err := strconv.Atoi(args[1])
	if err != nil {
		// Return an error if the argument is not a valid integer
		return &EvalResponse{
			Result: clientio.NIL,
			Error:  diceerrors.ErrIntegerOutOfRange,
		}
	}
	count = countArg
}

// Return the members with the lowest score
results = sortedSet.GetMin(count)

```

Reply:
args cannot be more than 2. So I am changing the condition from `len(args) >= 2` to `len(args) == 2`. Is this ok?

Reply:
Made the changes in this commit: [773521b](https://github.com/DiceDB/dice/pull/1143/commits/773521b261b16242553ac63b4a185801dca5cb87)